neurom/io/hdf5.py

```diff
@@ -184,29 +184,29 @@ def remove_duplicate_points(points, groups):
         '''
 
         group_initial_ids = groups[:, 0]
+        group_len = len(group_initial_ids)
 
         def _find_last_point(group_id):
             ''' Identifies and returns the id of the last point of a group'''
+            return group_initial_ids[group_id + 1] - 1
```

Comment:
According to my reading of the spec, the group IDs are supposed to be ordered like this. This doesn't translate to the branch order relationship you suggest, because the parent ordering isn't fixed. So `group_initial_ids[group_id + 1]` is not a child of `group_initial_ids[group_id]`, but its index is one more than the last point in `group_initial_ids[group_id]`.